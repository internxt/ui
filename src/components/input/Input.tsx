import { useState } from 'react';
import { Eye, EyeSlash } from '@phosphor-icons/react';

const Input = ({
  type = 'text',
  size = 'lg',
  disabled = false,
  accentColor,
  placeholder = '',
}: {
  type?: 'text' | 'password';
  size?: 'lg' | 'md';
  disabled?: boolean;
  accentColor?: 'red' | 'orange' | 'green';
  placeholder?: string;
}): JSX.Element => {
  const [intputType, setInputType] = useState(type);

  const handleInputType = () => {
    intputType === 'text' ? setInputType('password') : setInputType('text');
  };

  return (
    <div className="relative">
      <input
        type={intputType}
        disabled={disabled}
        placeholder={placeholder}
        className={`bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            ${size === 'lg' && 'text-lg h-10 px-4'}
            ${size === 'md' && 'text-base h-9 px-3.5'}
            ${!disabled ? 'border-gray-20 text-gray-100' : 'border-gray-5 text-gray-40'}
            ${!accentColor && 'border-gray-20 focus:border-primary focus:ring focus:ring-primary/10'}
            ${accentColor === 'red' && 'border-red focus:ring focus:ring-red/10'}
            ${accentColor === 'orange' && 'border-orange focus:ring focus:ring-orange/10'}
            ${accentColor === 'green' && 'border-green focus:ring focus:ring-green/10'}
        `}
      />

      {type === 'password' && (
        <button onClick={handleInputType}>
          {intputType === 'text' ? (
            <EyeSlash
              className={`absolute h-6 w-6 
                ${size === 'lg' && 'right-4 top-2'}
                ${size === 'md' && 'right-3.5 top-1.5'}
            `}
            />
          ) : (
            <Eye
              className={`absolute h-6 w-6 
                ${size === 'lg' && 'right-4 top-2'}
                ${size === 'md' && 'right-3.5 top-1.5'}
            `}
            />
          )}
        </button>
      )}
    </div>
  );
};

export default Input;
