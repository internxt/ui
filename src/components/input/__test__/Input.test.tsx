import React from 'react';
import { describe, expect, it } from 'vitest';
import Input from '../Input';
import renderer from 'react-test-renderer';

describe('Button component', () => {
  it('Input disabled render correctly', () => {
    const button = renderer.create(<Input disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Input enabled should render correctly', () => {
    const button = renderer.create(<Input />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Input enabled with placeholder should render correctly', () => {
    const button = renderer.create(<Input placeholder="Placeholder" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Input enabled with accent color red should render correctly', () => {
    const button = renderer.create(<Input accentColor="red" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Input enabled with accent color orange should render correctly', () => {
    const button = renderer.create(<Input accentColor="orange" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Input enabled with accent color green should render correctly', () => {
    const button = renderer.create(<Input accentColor="green" />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
