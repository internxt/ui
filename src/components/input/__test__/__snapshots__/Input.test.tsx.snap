// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > Input disabled render correctly 1`] = `
<div>
  <input
    className="bg-transparent border rounded-md outline-none text-base font-regular placeholder:text-gray-30
        h-10 px-4
        false
        border-gray-5 text-gray-40
        border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
        false
        false
        false
        "
    disabled={true}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input enabled should render correctly 1`] = `
<div>
  <input
    className="bg-transparent border rounded-md outline-none text-base font-regular placeholder:text-gray-30
        h-10 px-4
        false
        border-gray-5 text-gray-100
        border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
        false
        false
        false
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input enabled with accent color green should render correctly 1`] = `
<div>
  <input
    className="bg-transparent border rounded-md outline-none text-base font-regular placeholder:text-gray-30
        h-10 px-4
        false
        border-gray-5 text-gray-100
        false
        false
        false
        border-green focus:ring focus:ring-green/10
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input enabled with accent color orange should render correctly 1`] = `
<div>
  <input
    className="bg-transparent border rounded-md outline-none text-base font-regular placeholder:text-gray-30
        h-10 px-4
        false
        border-gray-5 text-gray-100
        false
        false
        border-orange focus:ring focus:ring-orange/10
        false
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input enabled with accent color red should render correctly 1`] = `
<div>
  <input
    className="bg-transparent border rounded-md outline-none text-base font-regular placeholder:text-gray-30
        h-10 px-4
        false
        border-gray-5 text-gray-100
        false
        border-red focus:ring focus:ring-red/10
        false
        false
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input enabled with placeholder should render correctly 1`] = `
<div>
  <input
    className="bg-transparent border rounded-md outline-none text-base font-regular placeholder:text-gray-30
        h-10 px-4
        false
        border-gray-5 text-gray-100
        border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
        false
        false
        false
        "
    disabled={false}
    placeholder="Placeholder"
    type="text"
  />
</div>
`;

exports[`Button component > Input type password enabled render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-20 text-gray-100
            border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
            false
            false
            false
        "
    disabled={false}
    placeholder=""
    type="password"
  />
  <button
    onClick={[Function]}
  >
    <svg
      className="absolute h-6 w-6 
                right-4 top-2
                false
            "
      fill="none"
      height="15"
      viewBox="0 0 15 15"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`Button component > Input type password enabled should render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-20 text-gray-100
            border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
            false
            false
            false
        "
    disabled={false}
    placeholder=""
    type="password"
  />
  <button
    onClick={[Function]}
  >
    <svg
      className="absolute h-6 w-6 
                right-4 top-2
                false
            "
      fill="none"
      height="15"
      viewBox="0 0 15 15"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`Button component > Input type text disabled render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-5 text-gray-40
            border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
            false
            false
            false
        "
    disabled={true}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input type text enabled should render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-20 text-gray-100
            border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
            false
            false
            false
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input type text enabled with accent color green should render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-20 text-gray-100
            false
            false
            false
            border-green focus:ring focus:ring-green/10
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input type text enabled with accent color orange should render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-20 text-gray-100
            false
            false
            border-orange focus:ring focus:ring-orange/10
            false
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input type text enabled with accent color red should render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-20 text-gray-100
            false
            border-red focus:ring focus:ring-red/10
            false
            false
        "
    disabled={false}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`Button component > Input type text enabled with placeholder should render correctly 1`] = `
<div
  className="relative"
>
  <input
    className="bg-transparent border rounded-md outline-none font-regular placeholder:text-gray-30
            text-lg h-10 px-4
            false
            border-gray-20 text-gray-100
            border-gray-20 focus:border-primary focus:ring focus:ring-primary/10
            false
            false
            false
        "
    disabled={false}
    placeholder="Placeholder"
    type="text"
  />
</div>
`;
